ark/utils/data_utils.py

```diff
@@ -4,6 +4,77 @@
 import numpy as np
 import xarray as xr
 
+from ark import settings
+from ark.utils.misc_utils import verify_in_list
+
+
+def label_cells_by_cluster(fovs, all_data, label_maps, fov_col=settings.FOV_ID,
+                           cell_label_column=settings.CELL_LABEL,
+                           cluster_column=settings.KMEANS_CLUSTER):
+    """ Translates cell-ID labeled images according to the clustering assignment.
+
+    Takes a list of fovs, and relabels each image (array) according to the assignment
+    of cell IDs to cluster label.
+
+    Args:
+        fovs (list):
+            List of fovs to relabel.
+        all_data (pandas.DataFrame):
+            data including fovs, cell labels, and cell expression matrix for all markers.
+        label_maps (xr.DataArray):
+            xarray of label maps for multiple fovs
+        fov_col (str):
+            column with the fovs names in all_data.
+        cell_label_column (str):
+            column with the cell labels in all_data.
+        cluster_column (str):
+            column with the cluster labels in all_data.
+    Returns:
+        xr.DataArray:
+            The relabeled images (dims: ["fovs", "rows", "cols"]).
+    """
+
+    # check if included fovs found in fov_col
+    verify_in_list(fov_names=fovs, unique_fovs=all_data[fov_col].unique())
+    verify_in_list(fov_names=fovs, unique_fovs=label_maps.fovs.values)
```

Comment:
The way the verify function works is that when it fails, it reports the name of the argument that didn't match. So it helps to give the variables more informative names, so that the error message will help to identify the problem area. Right now, the same error message will be raised if the FOV isn't in label_map, or if the FOV isn't in all data. It would be better to rename `unique_fovs` so that it reflects where the fovs are coming from, i.e. `label_map_fovs`, `extracted_data_fovs`, or something like that

Reply:
Changed to `label_map_fovs` and `all_data_fovs`